docs/src/literate/topology_optimization.jl

```diff
@@ -0,0 +1,503 @@
+# # Topology optimization
+#
+# **Keywords**: *Topology optimization*, *weak and strong form*, *non-linear problem*, *Laplacian*, *grid topology*
+#
+# ![](bending_animation.gif)
+#
+# *Figure 1*: Optimization of the bending beam. Evolution of the density for fixed total mass.
+#
+#-
+#md # !!! tip
+#md #     This example is also available as a Jupyter notebook:
+#md #     [`topology_optimization.ipynb`](@__NBVIEWER_ROOT_URL__/examples/topology_optimization.ipynb).
+#-
+#
+# ## Introduction
+#
+# Topology optimization is the task of finding structures that are mechanically ideal. 
+# In this example we cover the bending beam, where we specify a load, boundary conditions and the total mass. Then, our
+# objective is to find the most suitable geometry within the design space minimizing the compliance (i.e. the inverse stiffness) of the structure.
+# We shortly introduce our simplified model for regular meshes. A detailed derivation of the method and advanced techniques 
+# can be found in [Article 1](https://onlinelibrary.wiley.com/doi/full/10.1002/nme.5988) and
+# [Article 2](https://www.sciencedirect.com/science/article/abs/pii/S0045782522000755?via%3Dihub). 
+#
+# We start by introducing the local, elementwise density $\chi \in [\chi_{\text{min}}, 1]$ of the material, where we choose
+# $\chi_{\text{min}}$ slightly above zero to prevent numerical instabilities. Here, $\chi = \chi_{\text{min}}$ means void and $\chi=1$
+# means bulk material. Then, we use a SIMP ansatz (solid isotropic material with penalization) for the stiffness tensor
+# $C(\chi) = \chi^p C_0$, where $C_0$ is the stiffness of the bulk material. The SIMP exponent $p>1$ ensures that the
+# model prefers the density values void and bulk before the intermediate values. The variational formulation then yields
+# the modified Gibbs energy 
+# ```math
+# G = \int_{\Omega} \frac{1}{2} \chi^p \varepsilon : C : \varepsilon \; \text{d}V - \int_{\Omega} \boldsymbol{f} \cdot \boldsymbol{u} \; \text{d}V - \int_{\partial\Omega} \boldsymbol{t} \cdot \boldsymbol{u} \; \text{d}A.
+# ```
+# which we will solve with the finite element method using Ferrite. Furthermore, we receive the evolution equation of the density 
```

Comment:
Wording. We do not "solve" the energy with the finite element method.

Reply:
```suggestion
# Furthermore, we receive the evolution equation of the density in the strong form as
```